### PR TITLE
feat: add benchmark script

### PR DESCRIPTION
**Pull request by Himself65**

### CODE REVIEW

**Comment by Himself65** [comment]
```bash
Cold processing
┌──────────────────────────────┬───────────────┐
│           (index)            │ Cost time (s) │
├──────────────────────────────┼───────────────┤
│ Load Plugin/Scripts/Database │    '0.38s'    │
│        Process Source        │    '7.94s'    │
│         Render Files         │    '8.66s'    │
│        Save Database         │    '0.34s'    │
│          Total time          │   '17.32s'    │
└──────────────────────────────┴───────────────┘
Hot processing
┌──────────────────────────────┬───────────────┐
│           (index)            │ Cost time (s) │
├──────────────────────────────┼───────────────┤
│ Load Plugin/Scripts/Database │    '1.18s'    │
│        Process Source        │    '1.31s'    │
│         Render Files         │    '8.10s'    │
│        Save Database         │    '0.52s'    │
│          Total time          │   '11.11s'    │
└──────────────────────────────┴───────────────┘
Another Cold processing
┌──────────────────────────────┬───────────────┐
│           (index)            │ Cost time (s) │
├──────────────────────────────┼───────────────┤
│ Load Plugin/Scripts/Database │    '0.40s'    │
│        Process Source        │    '8.00s'    │
│         Render Files         │    '8.27s'    │
│        Save Database         │    '0.32s'    │
│          Total time          │   '16.99s'    │
└──────────────────────────────┴───────────────┘
```

File: script/benchmark.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  } else {
+    spawnSync('ln', [
+      '-sf',
+      path.resolve(__dirname, '..'),
+      path.resolve(testDir, 'node_modules', 'hexo')
+    ]);
+  }
+}
+
+function deleteFolderRecursive(pth) {
```
Comment (Himself65) [inline script/benchmark.js:124]: compatibility with node10.x

Comment (SukkaW) [reply script/benchmark.js:124]: `hexo-fs` has [`emptyDir`](https://github.com/hexojs/hexo-fs#emptydirpath-options) method.

File: script/benchmark.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
+'use strict';
+
+const { performance, PerformanceObserver } = require('perf_hooks');
+const { spawnSync, spawn } = require('child_process');
```
Comment (SukkaW) [inline script/benchmark.js:4]: hexo-util has [`spawn`](https://github.com/hexojs/hexo-util#spawncommand-args-options) method

File: .github/workflows/tester.yml [hunk 3]
```diff
@@ -34,6 +34,8 @@ jobs:
           github-token: ${{ secrets.github_token }}
           flag-name: run-${{ runner.os }}-${{ matrix.node-version }}
           parallel: true
+      - name: Benchmark
+        run: node script/benchmark.js
```
Comment (SukkaW) [inline .github/workflows/tester.yml:38]: Currently we run benchmark on Travis CI: https://github.com/hexojs/hexo/blob/master/.travis.yml

Would you mind adding the benchmark script in `.travis.yml` instead?

File: script/benchmark.js (hunk 2 above)
Comment (Himself65) [reply script/benchmark.js:4]: I see that, but it's returns a promise which I couldn't add listener on it

File: script/benchmark.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+      performance.measure('Render Files', 'file-loaded', 'generated');
+      performance.measure('Save Database', 'generated', 'database-saved');
+      performance.measure('Total time', 'hexo-begin', 'database-saved');
+      obs.disconnect();
+      resolve();
+    });
+
+    process.once('exit', () => {
+      console.log(name);
+      console.table(result);
```
Comment (SukkaW) [inline script/benchmark.js:67]: https://github.com/hexojs/hexo/blob/75201130fef4a02f27e239200714d5815f97ee3a/test/benchmark.sh#L39-L44

Performance regression detection could be added (simply a threshold of `20s` will be fine).

`chalk` could be utilized for colorize output.

**Pull request merged**